tsconfig.samples.json

```diff
@@ -0,0 +1,10 @@
+{
```

Comment:
is it possible to have a better file name since the options here are not specific to samples. some ideas: "typecheck"/"dev"/"noemit"/"nodenext"...

Reply:
I meant to indicate that it is meant to be used for code written on top of our libraries and not for library code itself. I think none of the proposed name convey that and could add to the confusion.

Reply:
"non-lib"/"non-dist"?

Reply:
Ok, I adopted `nonlib` in https://github.com/Azure/azure-sdk-for-js/pull/32066/commits/e35d324a2f29a5cbb3b66019868b89979837a2fa